app/models/katello/package_group.rb

```diff
@@ -1,7 +1,30 @@
 module Katello
-  class PackageGroup
+  class PackageGroup < Katello::Model
+    include Glue::Pulp::PulpContentUnit
     include Glue::Pulp::PackageGroup if Katello.config.use_pulp
-    include Glue::ElasticSearch::PackageGroup if Katello.config.use_elasticsearch
+
     CONTENT_TYPE = "package_group"
+
+    has_many :repositories, :through => :repository_package_groups, :class_name => "Katello::Repository"
```

Comment:
The data modelling issue is present in pulp as well.  My biggest concern would be to mirror what is in pulp rather than to try to add our own validations on top of it.  Otherwise we could get into a situation where pulp may specify that a group belongs to 2 repos and our indexing/syncing gets broken because of it.